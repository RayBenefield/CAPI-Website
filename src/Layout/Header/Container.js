import { connect } from 'react-redux';
import Header from '.';

const mapStateToProps = () => ({});
const mapDispatchToProps = (dispatch) => {
    return {
        onAddClick: () => {
            dispatch({ type: 'LOAD_FILES' })
        }
    }
}

const HeaderContainer = connect(
    mapStateToProps,
    mapDispatchToProps
)(Header);

export default HeaderContainer;
