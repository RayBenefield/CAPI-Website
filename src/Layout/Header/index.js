import React from 'react';
import Paper from 'material-ui/Paper';
import Fab from 'material-ui/FloatingActionButton';
import ContentAdd from 'material-ui/svg-icons/content/add';
import logo from './logo.svg';
import './index.css';

const Header = ({ id, onAddClick, header }) => (
    <Paper
        id={id}
        className="App-header"
        zDepth={5}
        style={{
            height: `${header.height}px`,
            boxSizing: 'content-box',
            backgroundColor: '#212121',
            zIndex: 5,
            position: 'relative',
        }}
    >
        <div
            style={{
                width: '100%',
                position: 'absolute',
                bottom: '60px',
            }}
        >
            <img
                src={logo}
                className="App-logo"
                alt="logo"
            />
        </div>
        <h2
            style={{
                position: 'absolute',
                width: '100%',
                bottom: '0px',
                left: '0px',
            }}
        >Halo File Management</h2>
        <Fab
            zDepth={4}
            style={{
                position: 'absolute',
                top: `${header.height - 28}px`,
                right: '50px',
            }}
            onClick={() => onAddClick()}
        >
            <ContentAdd />
        </Fab>
    </Paper>
);

export default Header;
