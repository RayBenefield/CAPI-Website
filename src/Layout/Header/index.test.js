import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Header from '.';

it('renders without crashing', () => {
    const wrapper = shallow(<Header />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
});
