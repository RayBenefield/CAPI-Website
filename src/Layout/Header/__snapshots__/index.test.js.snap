exports[`test renders without crashing 1`] = `
<Paper
  circle={false}
  className="App-header"
  id={undefined}
  rounded={true}
  style={
    Object {
      "backgroundColor": "#212121",
      "boxSizing": "content-box",
      "position": "relative",
      "zIndex": 5,
    }
  }
  transitionEnabled={true}
  zDepth={5}>
  <img
    alt="logo"
    className="App-logo"
    src="test-file-stub" />
  <h2>
    Halo File Management
  </h2>
  <FloatingActionButton
    disabled={false}
    mini={false}
    onClick={[Function]}
    secondary={false}
    style={
      Object {
        "position": "absolute",
        "right": "50px",
        "top": "165px",
      }
    }
    zDepth={4}>
    <ContentAdd />
  </FloatingActionButton>
</Paper>
`;
